### DISCUSSION TITLE
How to rotate labels in ValueAxis or DateAxis

### DISCUSSION THREAD

**Discussion by peterkronenberg** [jfree/jfreechart#372] [General]
I"ve got a time series chart and I want to be able to rotate the labels like this
![image](https://github.com/jfree/jfreechart/assets/5097682/98a90e7f-d929-47ee-b41d-3bd7ef2ac50d)

I found this post from many years ago, https://www.jfree.org/forum/viewtopic.php?t=25189, which seems to work, but the problem is that it eliminates the axis label.  Here the before and after

Note the x-axis is labeled **Date**
![image](https://github.com/jfree/jfreechart/assets/5097682/ba8ba970-58ac-4b0b-9941-7f0b759408f2)

After applying the new code, I get this
![image](https://github.com/jfree/jfreechart/assets/5097682/1380d014-c985-4a16-9974-d2699baad3ea)

The axis label is missing.

Is there a better solution?

**Comment by trashgod** [comment]
[`DateAxis`](https://github.com/jfree/jfreechart/blob/master/src/main/java/org/jfree/chart/axis/DateAxis.java) relies on the parent's `isVerticalTickLabels()` to reserve space for the axis label. I suspect your custom tick labels are simply covering the axis label. What happens if you enable vertical tick labels?

**Comment by peterkronenberg** [comment]
Works the same with and without `isVerticalTickLabels()`.  Also looked to see if there was some padding option, but couldn't find anything

**Reply by trashgod** [reply]
I've posted a reference example [here](https://stackoverflow.com/a/76585316/230513).
